### PR TITLE
use options as attributes

### PR DESCRIPTION
**Pull request by vegichan**
Pass options such as LBPolicy to the admin

I can't test the LBPolicy with the current Travis setup. It works locally.

### CODE REVIEW

**Comment by vegichan** [comment]
The respective C++ code is

`Connect connect(tlsContext, "SQL2");

        const char* passwd = "";
        account = "";
        bool directConnection = false;

        const char* ciphers = CryptoSocket::getDefaultCiphers();

        IpVersionOption ipVers = IpVersionV4;

        // TLS configuration parameters
        bool verifyHostname = true;
        bool srpFallback = false;
        const char* trustStore = nullptr;
        const char* trustedCerts = nullptr;

        int nparams = connectionParameters.getCount();
        bool addCiphers = true;
        for (int n = 0; n < nparams; ++n) {
            const char* property = connectionParameters.getName(n);
            const char* value = connectionParameters.getValue(n);

            if (strcmp(property, "password") == 0) {
                passwd = value;
            } else if (strcmp(property, "user") == 0) {
                account = value;
            } else if (strcmp(property, "schema") == 0) {
                connect.addAttribute("Schema", value);
            } else if (strcmp(property, "ciphers") == 0) {
                ciphers = value;
                addCiphers = false;
            } else if (strcmp(property, "allowSRPFallback") == 0) {
                srpFallback = String::equalsNoCase(value, "true");
            } else if (strcmp(property, "verifyHostname") == 0) {
                verifyHostname = String::equalsNoCase(value, "true");
            } else if (strcmp(property, "trustStore") == 0) {
                trustStore = value;
            } else if (strcmp(property, "trustedCertificates") == 0) {
                trustedCerts = value;
            } else if (strcmp(property, "ipVersion") == 0) {
                if (strcmp(value, "v6") == 0) {
                    ipVers = IpVersionV6;
                }
            } else if (strcmp(property, "direct") == 0) {
                directConnection = String::equalsNoCase(value, "true");
            } else {
                connect.addAttribute(property, value);
            }
        }
`

**Comment by adriansuarez** [comment]
> Pass options such as LBPolicy to the admin
> 
> I can't test the LBPolicy with the current Travis setup. It works locally.

Isn't the issue that connection properties weren't being sent to the Admin at all? You can choose a connection property that nuoagent would process. You can look at some of the nuoagent load balancers to find a suitable one. I remember there being some load-balancer that did something with a connection property called `NodeKey`.

File: pynuodb/session.py [hunk 1]
```diff
@@ -139,6 +139,22 @@ def __init__(self, host, port=None, service="Identity", timeout=None,
                 else:
                     raise
 
+    @staticmethod
+    def _extract_options(options):
+        expected_extracted_options = ['password', 'user',
+                             'trustStore', 'verifyHostname', 'allowSRPFallback', 'ciphers',
+                             'ipVersion', 'direct']
```
Comment (adriansuarez) [inline pynuodb/session.py]: Do we have a reference implementation of the driver that we base changes like this on? What do JDBC or the C++ driver?

Comment (vegichan) [reply pynuodb/session.py]: I copy pasted that to this review as a comment

**Pull request merged**